src/comments.js

```diff
@@ -412,19 +416,27 @@ function printComments(path, print, options) {
             parent
           )
         );
+      } else {
+        // Those two node types print their dangling comments themselves
+        if (value.type !== "Program" && value.type !== "BlockStatement") {
+          trailingParts.push(
+            printDanglingComment(
+              commentPath,
+              print,
+              options
+            )
+          );
+        }
```

Comment:
JSXExpressionContainer never returns `"{}"`, it always call `path.call(print, "expression")` which is never empty because there's always a `JSXEmptyExpression`. So it prints correctly in this case.

For things where we return `"{}"`, it prints after. It's not ideal but at least it prints it. There's an example in the snapshots

![image](https://cloud.githubusercontent.com/assets/197597/22384529/9f15c328-e483-11e6-907a-5b31de24bf22.png)